### PR TITLE
i18n(ja): Add Japanese `guides/troubleshooting.md`

### PR DESCRIPTION
**Pull request by kyosuke**
#### What kind of changes does this PR include?

- Translated content

#### Description

- Translated the troubleshooting page into Japanese.
- Fixed a link on `migrate.md`
- Updated sidebar navigation

### CODE REVIEW

File: src/pages/ja/guides/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+```yaml
+nodeLinker: "node-modules"
+```
+
+## ヒントとコツ
+
+### `console.log()`を使ったデバッグ
+
+`console.log()`は、Astroのコードをデバッグするためのシンプルでありながら人気のある方法です。console.log`ステートメントをどこに書くかによって、デバッグ出力がどこに出力されるかが決まります。
```
Comment (yuto343) [inline src/pages/ja/guides/troubleshooting.md:145]:
```suggestion
`console.log()`は、Astroのコードをデバッグするためのシンプルでありながら人気のある方法です。`console.log`ステートメントをどこに書くかによって、デバッグ出力がどこに出力されるかが決まります。
```

**Review by yuto343 — COMMENTED** [review]
@kyosuke レビューさせてもらいました！

File: src/pages/ja/guides/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+Astroは、コードのトラブルシューティングとデバッグを支援するため、いくつかのツールを提供しています。
+
+## よくあるエラーメッセージ
+
+ターミナルで表示される一般的なエラーメッセージとその意味、および対処法を紹介します。
+
+### Transform failed with X error （X エラーで変換に失敗しました）
+
+このメッセージは、インポートとエクスポートのステートメントが `.astro` ファイルの先頭にあることを要求するAstroの現在の制限のためによく表示されます。
```
[the remark this answers is not in the document]
Comment (kyosuke) [reply src/pages/ja/guides/troubleshooting.md:16]: なるほど、確かにわかりづらいので、いただいた案を参考に、こんな感じに直してみました。

```suggestion
現在のAstroの制限として、インポートとエクスポートのステートメントが`.astro`ファイルの先頭になければならないため、このメッセージが表示されることがあります。
```

File: src/pages/ja/guides/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+#### そうではない？
+
+**解決方法**: [Astroコンポーネント](/ja/core-concepts/astro-components/)または[UIフレームワークコンポーネント](/ja/core-concepts/framework-components/)の適切なドキュメントをチェックする。[Astro.new](https://astro.new)からAstroのスターターテンプレートを開き、最小限のAstroプロジェクトでコンポーネントのみのトラブルシューティングを検討してください。
+
+**自分だけの問題ではないと思うなら？**  
+[この問題](https://github.com/withastro/astro/issues?q=is%3Aissue+is%3Aopen+Unable+to+render+Component)を報告した人が他にいないか確認してください！
+
+### Expected a default export（デフォルトのエクスポートを予想）
+
+このエラーは、無効なコンポーネントをインポートまたはレンダリングしようとしたとき、または正しく動作していないコンポーネントをインポートまたはレンダリングしようとしたときにスローされることがあります。(この特定のメッセージは、AstroでのUIコンポーネントのインポートが動作する方法によって発生します。)
```
[the remark this answers is not in the document]
Comment (kyosuke) [reply src/pages/ja/guides/troubleshooting.md:57]:
```suggestion
このエラーは、無効なコンポーネントをインポートまたはレンダリングしようとしたとき、または正しく動作していないコンポーネントをインポートまたはレンダリングしようとしたときに表示されることがあります。(この特定のメッセージは、AstroでのUIコンポーネントのインポートが動作する方法によって発生します。)
```

**Comment by kyosuke** [comment]
@yuto343 レビューありがとうございます！コメントいただいた場所修正しました！お手数ですが、再度確認お願いします。

File: src/pages/ja/guides/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+### コンポーネントがレンダリングされない
+
+まず、[`.astro`のコンポーネントスクリプト](/ja/core-concepts/astro-components/#コンポーネントスクリプト)または[`.md`のfront-matter](/ja/guides/markdown-content/#markdownでコンポーネントを使用する)で**コンポーネントをインポート**しているかどうか確認します。
+
+それから、import文を確認してください。
+
+- インポートのリンク先が違っていませんか？（importパスを確認してください）
+- インポートしたコンポーネントと同じ名前になっていますか？(コンポーネント名と、[`.Astro`構文にしたがっていること](/ja/comparing-astro-vs-other-tools/#astro-vs-jsx)を確認してください。）
+- インポート時に拡張子が含まれていますか？(インポートしたファイルに拡張子が含まれているか確認してください。例: `.Astro`, `.md`, `.jsx`, `.vue`)
```
Comment (kyosuke) [inline src/pages/ja/guides/troubleshooting.md:74]:
```suggestion
- インポート時に拡張子が含まれていますか？（インポートしたファイルに拡張子が含まれているか確認してください。例: `.Astro`、`.md`、`.jsx`、`.vue`、`.svelte`。 注: `.js(x)`と`.ts(x)`のファイルのみ、拡張子は必要ありません）
```

https://github.com/withastro/docs/commit/c72ed8bdd3fa5631f1639f6529dfdb595ae2b456

**Comment by kyosuke** [comment]
https://github.com/withastro/docs/commit/c72ed8bdd3fa5631f1639f6529dfdb595ae2b456 の変更も反映しました。

File: src/pages/ja/guides/troubleshooting.md [hunk 5]
```diff
@@ -13,7 +13,7 @@ Astroは、コードのトラブルシューティングとデバッグを支援
 
 ### Transform failed with X error （X エラーで変換に失敗しました）
 
-このメッセージは、インポートとエクスポートのステートメントが `.astro` ファイルの先頭にあることを要求するAstroの現在の制限のためによく表示されます。
+現在のAstroの制限として、インポートとエクスポートのステートメントが`.astro`ファイルの先頭になければならないため、このメッセージが表示されることがあります。
```
Comment (yuto343) [inline src/pages/ja/guides/troubleshooting.md:16]: 👍 👍

**Review by yuto343 — APPROVED** [review]
LGTM！

**Comment by Yan-Thomas** [comment]
Amazing work @kyosuke & @yuto343! Pointing out that we just pushed some changes to the troubleshooting page that aren't addressed in this PR, you just have to remove this section and I'll merge it. Thanks 🙌!  https://github.com/withastro/docs/commit/1c900121aa0c5b91761c4cd99997a692312510a6

**Comment by kyosuke** [comment]
@Yan-Thomas Thanks! I removed that section!

**Review by delucis — APPROVED** [review]
Thank you @kyosuke! 🙌 
And thank you @yuto343! 🙌

**Pull request merged**